app/models/resource_type.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class ResourceType < ActiveRecord::Base
+  validates :name, presence: true, uniqueness: true
+  validates :dtd_file, presence: true
+
+  after_validation :dtd_file_exists
+
+  private
+
+  def dtd_file_exists
+    return if File.exist?("public/xmlns/#{dtd_file}")
+    raise "ResourceType with name: #{name} does not have valid DTD file in 'public/xmlns/'."
```

Comment:
Wondering if this message might be a bit misleading in that it says there's no a 'valid' DTD file. Doesn't seem like any validation of the contents of the DTD file is happening here.